addons/web/static/tests/core/widget_tests.js

```diff
@@ -377,16 +379,49 @@ QUnit.module('core', {}, function () {
     });
 
     QUnit.test("calling _rpc on destroyed widgets", function (assert) {
-        assert.expect(1);
+        assert.expect(3);
+
+        var def;
+        var parent = new Widget();
+        testUtils.addMockEnvironment(parent, {
+            session: {
+                rpc: function () {
+                    def = $.Deferred();
+                    def.abort = def.reject;
+                    return def;
+                },
+            },
+            services: {
+                ajax: AjaxService,
+            }
+        });
+        var widget = new Widget(parent);
 
-        var widget = new Widget();
+        widget._rpc({route: '/a/route'}).then(function () {
+            assert.ok(true, "The ajax call should be resolve");
+        });
+        def.resolve();
+        def = null;
+
+        widget._rpc({route: '/a/route'}).always(function () {
+            throw Error("Calling _rpc on a destroyed widget should return a " +
+                "deferred that is never resolved nor rejected");
```

Comment:
I don't understand the message: widget is not destroyed yet

Reply:
We destroy the widget just after, here the exception allows to see that the request is not done if the widget is destroyed.